internal/utils/http/http_test.go

```diff
@@ -0,0 +1,286 @@
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/covid19cz/erouska-backend/internal/utils/errors"
+	"github.com/stretchr/testify/assert"
+	rpccode "google.golang.org/genproto/googleapis/rpc/code"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type testRequest struct {
+	Field1 string `json:"field1" validate:"required"`
+	Field2 int    `json:"field2" validate:"required"`
+}
+
+type testResponse struct {
+	Result string `json:"result"`
+}
+
+/* DecodeJSONBody: */
+
+func TestDecodeJSONBodyOk(t *testing.T) {
+	var body = bytes.NewBufferString(`{"data": {"field1": "ahoj", "field2": 42}}`)
+
+	req, err := http.NewRequest("GET", "/Url", body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	rr := httptest.NewRecorder()
+
+	var request testRequest
+
+	err = DecodeJSONBody(rr, req, &request)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, request, testRequest{
+		Field1: "ahoj",
+		Field2: 42,
+	})
+
+	assert.Equal(t, 0, rr.Body.Len())
```

Comment:
I don't think it is common to use assert in unit test in Golang. Given all the literature and real-life Golang projects I've always seen it done this way https://github.com/covid19cz/erouska-backend/blob/develop/internal/functions/registerehrid/register_ehrid_test.go#L21 using "tables" and cmp.Diff.

Reply:
I don't think this is wrong or anything. Even the test error output looks quite similar to cmp.Diff and it is easy to read. It's just weird how both of us are using different approaches and the codebase is incosistent :D

Reply:
I kind of hate _your_ approach as you still have to manually call the `panic` whereas `assert` does it for you. The code is already ugly and I don't like to make it even hairier and uglier :-( On the other hand, I agree that inconsistency is not ideal. What is `assert` for then, if not for testing?

Reply:
what is `goto` for then, if not for using it ever? :D 
I'm not calling `panic` manually. Why would I call it? Go test handles if it should continue or not.
I'll look later into what are the best practices. You can merge it now:)

Reply:
What I see is "(1) get me the diff and (2) if it's non-empty, fail with message I had to write" instead of "(1) fail if not equal":
```go
diff := cmp.Diff(ehrid, table.y())
if diff != "" {
	t.Fatalf("register mismatch (-want +got):\n%v", diff)
}
```
vs.
```go
assert.Equal(t, ehrid, table.y())
```

BTW good point with the GOTO, I wonder why it's even in the language when it's discouraged to use it (every language has sth like this, but Go is IMHO young enough to learn from the past...).